tests/baselines/reference/assignmentCompatWithObjectMembers.types

```diff
@@ -189,16 +189,16 @@ module ObjectTypes {
 >b : { foo: any; }
 
     var a2 = { foo: a2 };
->a2 : any
+>a2 : error

```

Comment:
I am somewhat suspicious of this; this file does not appear to have an error baseline, yet we are seeing the error type in the type baselines. Is there a missing error, or do we actually need to be returning any sometimes?

Reply:
It's about this case:

```ts
// @noImplicitAny: false
module ObjectTypes {
    var a2 = { foo: a2 };
}
```

I'm not sure what should be done about this. `errorType` does behave like `any`... but I'd be worried about some accidental propagation of it since it could be infectious for consumers. OTOH, `any` is already infectious.

Returning a different value based on `noImplicitAny` would kinda defeat the goal of this PR. Unless we'd also return the same value (based on `noImplicitAny`) right away when the circularity is detected for the first time (currently `errorType` is returned from there and it's often "replaced" by `anyType` when the stack unwinds).

Reply:
Looking again, I see a number of tests which have `: error` in their baselines but have no errors, so this is probably fine honestly.